### ISSUE TITLE
Feature: support OS keyrings

### ISSUE THREAD

**Issue by anatoli26** [keepassxreboot/keepassxc#115]
Subj. Here some brief intro into Gnome Keyring (GK) philosophy: https://wiki.gnome.org/Projects/GnomeKeyring/SecurityPhilosophy.

The idea is to support the OS keyrings (available on the 3 OSes) to unlock the DB. Basically, this is somehow a simplification of a scenario when the user account and the DB are locked with the same password.

Once the feature to lock the DB on session lock/termination (#100) is implemented, this may work quite efficiently: KPXR reads the DB pass from the keyring, unlocks it silently. When user's session locks (either on manual lock or via an inactivity timeout), the DB also locks, so no passwords are in the RAM (at least for GK). Once the user unlocks his/her account, the DB unlocks as well. This could be in combination with a keyfile. GK implementation details: https://wiki.gnome.org/Projects/GnomeKeyring/KeyringIntro

I believe this approach will improve the security for an average user, as for a passive attacker the obstacle is the same as a manually entered pass (or even bigger as the underlying DB pass (i.e. the actual pass with which the DB is locked) could be much more complex as it's entered only once, could be saved in some secret place and only used later on OS recovery, so no need to remember it).

At the same time, for an active attacker it's the same as he/she can install a keylogger or sniff the memory and get everything in any case (see the GK philosophy above).

The intention behind this proposal is to alleviate the suffering related to strong security for ordinary users and at the same time reduce the human factor (use of weak passwords) as it's impossible to make the ordinary users remember 3 strong different passwords (full-disk-encryption, user session with profile folder encrypted and pass manager).

**Comment by Manko10** [comment]
What I'd like to see is support for gpg-agent. My KWallet (which really only stores my ssh and owncloud passwords) is unlocked via gpg-agent + Yubikey, why not have the same for KeePassX. :-)

**Comment by phoerious** [comment]
Closing this in favor of #440 which provides direct suggestions how to implement OS keyring integration.
